### PR TITLE
Add descriptions to sync node and AI controller

### PR DESCRIPTION
**Pull request by metinc**
This adds descriptions to the props to make the plugin more accessible to new users.

Examples:

image

image

### CODE REVIEW

File: addons/godot_rl_agents/controller/ai_controller_2d.gd [hunk 1]
```diff
@@ -1,9 +1,17 @@
 extends Node2D
 class_name AIController2D
 
-enum ControlModes { INHERIT_FROM_SYNC, HUMAN, TRAINING, ONNX_INFERENCE, RECORD_EXPERT_DEMOS }
+enum ControlModes {
+    INHERIT_FROM_SYNC, ## Inherit setting from sync node
+    HUMAN, ## Test the environment manually
+    TRAINING, ## Train a model
+    ONNX_INFERENCE, ## Load a pretrained model using an .onnx file
+    RECORD_EXPERT_DEMOS
```
Comment (metinc) [inline addons/godot_rl_agents/controller/ai_controller_2d.gd:9]: Not sure what RECORD_EXPERT_DEMOS does so I don't have a description here.

File: addons/godot_rl_agents/sync.gd [hunk 2]
```diff
@@ -1,11 +1,21 @@
 extends Node
+class_name Sync
```
Comment (metinc) [inline addons/godot_rl_agents/sync.gd:2]: Somehow I had to add a class name to make the descriptions for ControlModes show up in the editor.

File: addons/godot_rl_agents/controller/ai_controller_2d.gd (hunk 1 above)
Comment (Ivan-267) [reply addons/godot_rl_agents/controller/ai_controller_2d.gd:9]: It's an option for recording expert demonstrations (i.e. playing the game manually to show the agent what to do). The data includes the obs and actions taken. However, a bit more code needs to be implemented when using this option.

There is a wip course bonus unit for this: https://github.com/huggingface/deep-rl-class/pull/537

Comment (metinc) [reply addons/godot_rl_agents/controller/ai_controller_2d.gd]: Thanks, I added the description.

File: addons/godot_rl_agents/sync.gd (hunk 2 above)
Comment (Ivan-267) [reply addons/godot_rl_agents/sync.gd:2]: I'm also not seeing the suggestions with Godot 4.3 beta 2 even with the class_name, I will test it more, maybe removing `.godot` folder will help after updating the plugin.

image

File: addons/godot_rl_agents/sync.gd [hunk 3]
```diff
@@ -1,11 +1,21 @@
...
 # --fixed-fps 2000 --disable-render-loop
 
-enum ControlModes { HUMAN, TRAINING, ONNX_INFERENCE }
+enum ControlModes {
+    HUMAN, ## Test the environment manually
+    TRAINING, ## Train a model
+    ONNX_INFERENCE ## Load a pretrained model using an .onnx file
+}
 @export var control_mode: ControlModes = ControlModes.TRAINING
+## Action will be repeated for n frames. Will introduce control lag if larger than 1.
```
Comment (Ivan-267) [inline addons/godot_rl_agents/sync.gd:12]:
```suggestion
## Action will be repeated for n frames (Godot physics steps). 
```

File: addons/godot_rl_agents/sync.gd [hunk 4]
```diff
@@ -1,11 +1,21 @@
...
+## Can be used to ensure that action_repeat on inference and training matches
```
Comment (Ivan-267) [inline addons/godot_rl_agents/sync.gd:13]:
```suggestion
```

File: addons/godot_rl_agents/sync.gd [hunk 5]
```diff
@@ -1,11 +1,21 @@
...
+## the recorded demonstrations.
```
Comment (Ivan-267) [inline addons/godot_rl_agents/sync.gd:14]:
```suggestion
```

File: addons/godot_rl_agents/controller/ai_controller_3d.gd [hunk 6]
```diff
@@ -1,9 +1,17 @@
...
-enum ControlModes { INHERIT_FROM_SYNC, HUMAN, TRAINING, ONNX_INFERENCE, RECORD_EXPERT_DEMOS }
+enum ControlModes {
+    INHERIT_FROM_SYNC, ## Inherit setting from sync node
+    HUMAN, ## Test the environment manually
+    TRAINING, ## Train a model
+    ONNX_INFERENCE, ## Load a pretrained model using an .onnx file
+    RECORD_EXPERT_DEMOS ## Record observations and actions for expert demonstrations
+}
 @export var control_mode: ControlModes = ControlModes.INHERIT_FROM_SYNC
+## The path to a trained .onnx model file to use for inference (only needed for the 'Onnx Inference' control mode).
```
Comment (Ivan-267) [inline addons/godot_rl_agents/controller/ai_controller_3d.gd:12]:
```suggestion
## The path to a trained .onnx model file to use for inference (overrides the path set in sync node).
```

File: addons/godot_rl_agents/controller/ai_controller_2d.gd [hunk 7]
```diff
@@ -1,9 +1,17 @@
...
-enum ControlModes { INHERIT_FROM_SYNC, HUMAN, TRAINING, ONNX_INFERENCE, RECORD_EXPERT_DEMOS }
+enum ControlModes {
+    INHERIT_FROM_SYNC, ## Inherit setting from sync node
+    HUMAN, ## Test the environment manually
+    TRAINING, ## Train a model
+    ONNX_INFERENCE, ## Load a pretrained model using an .onnx file
+    RECORD_EXPERT_DEMOS ## Record observations and actions for expert demonstrations
+}
 @export var control_mode: ControlModes = ControlModes.INHERIT_FROM_SYNC
+## The path to a trained .onnx model file to use for inference (only needed for the 'Onnx Inference' control mode).
```
Comment (Ivan-267) [inline addons/godot_rl_agents/controller/ai_controller_2d.gd:12]:
```suggestion
## The path to a trained .onnx model file to use for inference (overrides the path set in sync node).
```

**Review by Ivan-267 — APPROVED** [review]
Looks great overall, thanks for adding this. I've added a few small suggestions, after considering those feel free to merge the PR.

This may not work in some cases for the sync node in Godot without changing the class name as we discussed, but it will likely work in future versions.

**Comment by metinc** [comment]
Thanks! Feel free to merge it. I don't have write access on this repo.

**Pull request merged**